stdlib/encodings/charmap.pyi

```diff
@@ -0,0 +1,29 @@
+import codecs
+from _codecs import _CharMap
+from _typeshed import ReadableBuffer
+
+class Codec(codecs.Codec):
+    encode = codecs.charmap_encode  # type: ignore[assignment]  # pyright: ignore[reportAssignmentType]
```

Comment:
What are the errors here? Do type checkers handle these types correctly despite the ignores?

Reply:
stubtest:
```
stdlib/encodings/charmap.pyi:6: error: Incompatible types in assignment (expression has type "Callable[[str | None, dict[int, int] | _EncodingMap | None], tuple[bytes, int]]", base class "Codec" defined the type as "Callable[[str, str], tuple[bytes, int]]")  [assignment]
stdlib/encodings/charmap.pyi:7: error: Incompatible types in assignment (expression has type "Callable[[str | None, dict[int, int] | _EncodingMap | None], tuple[str, int]]", base class "Codec" defined the type as "Callable[[bytes, str], tuple[str, int]]")  [assignment]
```

It looks like mypy ends up with a type of Any here, which is too bad. The runtime source has a note:

```
    # Note: Binding these as C functions will result in the class not
    # converting them to methods. This is intended.
```

Which is why I didn't do an overriding method to start with. I'll update to that with a note.

Reply:
I confirmed that mypy handles the override type in this version correctly.